utils_cv/detection/model.py

```diff
@@ -0,0 +1,283 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+from typing import List, Tuple, Union
+from pathlib import Path
+from functools import partial
+
+from PIL import Image
+import cv2
+import torch
+import torch.nn as nn
+from torchvision import transforms
+from torchvision.models.detection import fasterrcnn_resnet50_fpn
+from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
+from torch.utils.data import Dataset, DataLoader
+import matplotlib.pyplot as plt
+
+from .references.transforms import RandomHorizontalFlip, Compose, ToTensor
+from .references.engine import train_one_epoch, evaluate
+from .references.utils import collate_fn
+from .references.coco_eval import CocoEvaluator
+from .plot import BoundingBoxParams, plot_boxes, plot_grid
+
```

Comment:
Same comment as before: if you copied (and edited) this code from somewhere, can you please provide a link.

Reply:
i wrote this class